pkg/kcp/provider/azure/redisinstance/createSubnet.go

```diff
@@ -0,0 +1,79 @@
+package redisinstance
+
+import (
+	"context"
+	"fmt"
+	"github.com/kyma-project/cloud-manager/api/cloud-control/v1beta1"
+	"github.com/kyma-project/cloud-manager/pkg/composed"
+	iprangeallocate "github.com/kyma-project/cloud-manager/pkg/kcp/iprange/allocate"
+	azureUtil "github.com/kyma-project/cloud-manager/pkg/kcp/provider/azure/util"
+	"github.com/kyma-project/cloud-manager/pkg/util"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func createSubnet(ctx context.Context, st composed.State) (error, context.Context) {
+	state := st.(*State)
+	logger := composed.LoggerFromCtx(ctx)
+
+	if state.subnet != nil {
+		return nil, nil
+	}
+
+	logger.Info("Creating Azure Redis subnet")
+
+	resourceGroupName := state.Scope().Spec.Scope.Azure.TechnicalID
+	virtualNetworkName := state.Scope().Spec.Scope.Azure.VpcNetwork
+	subnetName := azureUtil.GetSubnetName(state.Scope().Spec.Scope.Azure.TechnicalID)
+
+	existingRanges := []string{
+		state.Scope().Spec.Scope.Azure.Network.Nodes,
+		state.Scope().Spec.Scope.Azure.Network.Pods,
+		state.Scope().Spec.Scope.Azure.Network.Services,
+	}
+	cidr, error := iprangeallocate.AllocateCidr(22, existingRanges)
```

Comment:
This allocates a `/22` block, which is 1024 addresses. Are you sure we need that many?